src/generators/clientFileGenerator.ts

```diff
@@ -362,3 +436,135 @@ function writeClientOperations(
     })
   );
 }
+
+function getRequiredParamChecks(requiredParameters: ParameterDetails[]) {
+  return requiredParameters.map(
+    ({ name }) => `if(${name} === undefined) {
+    throw new Error("'${name}' cannot be null");
+  }`
+  );
+}
+
+function getCredentialScopesValue(credentialScopes?: string | string[]) {
+  if (Array.isArray(credentialScopes)) {
+    return `[${credentialScopes.map(scope => `"${scope}"`).join()}]`;
+  } else if (typeof credentialScopes === "string") {
+    return `"${credentialScopes}"`;
+  }
+
+  return credentialScopes;
+}
+
+function writeDefaultOptions(
+  hasCredentials: boolean,
+  hasLro: boolean,
+  clientDetails: ClientDetails
+) {
+  const { useCoreV2, credentialScopes, packageDetails } = getAutorestOptions();
+
+  const credentialScopesValues = getCredentialScopesValue(credentialScopes);
+  const addScopes = credentialScopes
+    ? `if(!options.credentialScopes) {
+    options.credentialScopes = ${credentialScopesValues}
+  }`
+    : "";
+
+  const defaults = !hasCredentials
+    ? `const defaults: ${clientDetails.className}OptionalParams = {
+    requestContentType: "application/json; charset=utf-8"
+  };`
+    : `const defaults: ${clientDetails.className}OptionalParams = {
+    requestContentType: "application/json; charset=utf-8",
+    credential: credentials
+  };`;
+
+  return !useCoreV2
+    ? `// Initializing default values for options
+  if (!options) {
+     options = {};
+  }
+
+  const defaultUserAgent = \`azsdk-js-\${packageName.replace(/@.*\\//,"")}/\${packageVersion} \${coreHttp.getDefaultUserAgentValue()}\`;
+  
+  ${addScopes}
+
+  super(${hasCredentials ? "credentials" : `undefined`}, {
+    ...options,
+    userAgent: options.userAgent
+      ? \`\${options.userAgent} \${defaultUserAgent}\`
+      : \`\${defaultUserAgent}\`
+  });
+  
+  this.requestContentType = "application/json; charset=utf-8";
+  
+  `
+    : `// Initializing default values for options
```

Comment:
NIT: Perhaps it will be more readable if each case is factored into its own function?

Reply:
Actually, I migrate them from ClientContextFileGenerator, but sure, let me change it.